sphinx/transforms/i18n.py

```diff
@@ -652,8 +652,10 @@ def apply(self, **kwargs: Any) -> None:
         if not self.config.translation_progress_classes:
             return
 
-        if self.config.translation_progress_classes is True:
+        if self.config.translation_progress_classes in {True, '1'}:
             add_translated = add_untranslated = True
+        elif self.config.translation_progress_classes in {False, '0'}:
+            add_translated = add_untranslated = False
```

Comment:
You can remove this and put it in the early return above:

```py
if self.config.translation_progress_classes in {False, '0'}:
	return
```

Reply:
Possibly - that'd modify the handling of `None` and empty-string values, though.

I'd suggest we allow the conversation about relocating the type-coercion to an earlier phase (during command-line parsing) to progress for a while and then revisit this if it remains relevant.